zam/aspirateur/zam_aspirateur/models.py

```diff
@@ -0,0 +1,51 @@
+import re
+from dataclasses import (
+    asdict,
+    dataclass,
+    replace,
+)
+from datetime import date
+from typing import Optional
+
+
+@dataclass
+class Amendement:
+
+    article: str        # libellé de l'article
+    alinea: str         # libellé de l'alinéa
+    num: str            # numéro d'amendement
+
+    auteur: str
+    groupe: Optional[str] = None        # groupe parlementaire
+
+    date_depot: Optional[date] = None
+
+    sort: Optional[str] = None          # retiré, adopté, etc.
+
+    discussion_commune: Optional[bool] = None
+    identique: Optional[bool] = None
+
+    dispositif: Optional[str] = None    # texte de l'amendement
+    objet: Optional[str] = None         # motivation
+
+    @property
+    def num_int(self) -> int:
+        """
+        Numéro d'amendement sous forme purement numérique (sans suffixes)
+        """
+        mo = re.search(r'(\d+)', self.num)
+        assert mo is not None
+        return int(mo.group(1))
+
+    def url(self, session, num_texte):
+        return "http://www.senat.fr/amendements/{}/{}/{}".format(
+            session,
+            num_texte,
+            self.html_page,
+        )
+
+    def evolve(self, **changes) -> 'Amendement':
```

Comment:
Why not `    evolve = replace` if it's for the name?

Reply:
I was a bit sloppy when refactoring from `attrs` to `dataclasses`. I thought I would get back to it later. I guess later is now. 😄

Reply:
Fixed in afb31d24ab9400aba81b20eee97a394074a8a6c9